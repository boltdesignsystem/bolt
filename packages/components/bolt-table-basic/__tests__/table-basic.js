import { render, stopServer } from '../../../testing/testing-helpers';
import schema from '../table-basic.schema';
const {
  format,
  borderless,
  // eslint-disable-next-line camelcase
  fixed_width_table,
  // eslint-disable-next-line camelcase
  fixed_width_columns,
} = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
}, 100);

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    rows: [
      {
        cells: ['R1C1', 'R1C2', 'R1C3'],
      },
      {
        cells: ['R2C1', 'R2C2', 'R2C3'],
      },
      {
        cells: ['R3C1', 'R3C2', 'R3C3'],
      },
    ],
  };

  const headerData = {
    headers: {
      top: {
        cells: ['Column 1', 'Column 2', 'Column 3'],
      },
    },
  };

  const sideHeaderData = {
    headers: {
      side: {
        cells: ['Row 1', 'Row 2', 'Row 3', 'Footer'],
      },
    },
  };

  const footerData = {
    footer: {
      cells: ['FC1', 'FC2', 'FC3'],
    },
  };

  fixtures = {
    defaultData,
    headerData,
    sideHeaderData,
    footerData,
  };
});

describe('Bolt Table Basic', () => {
  test(`default`, async () => {
    const results = await render(
      '@bolt-components-table-basic/table-basic.twig',
      {
        ...fixtures.defaultData,
      },
    );
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});

describe('Bolt Table Basic Component Prop -', () => {
  // // Target each of the schema keys with the following pattern
  format.enum.forEach(async option => {
    test(`format items: ${option}`, async () => {
      const results = await render(
        '@bolt-components-table-basic/table-basic.twig',
        {
          ...fixtures.defaultData,
          format: option,
        },
      );
      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });

  test(`borderless: true`, async () => {
    const results = await render(
      '@bolt-components-table-basic/table-basic.twig',
      {
        ...fixtures.defaultData,
        borderless: true,
      },
    );
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`fixed_width_table: true`, async () => {
    const results = await render(
      '@bolt-components-table-basic/table-basic.twig',
      {
        ...fixtures.defaultData,
        fixed_width_table: true,
      },
    );
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  fixed_width_columns.enum.forEach(async option => {
    test(`fixed_width_columns items: ${option}`, async () => {
      const results = await render(
        '@bolt-components-table-basic/table-basic.twig',
        {
          ...fixtures.defaultData,
          fixed_width_columns: option,
        },
      );
      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
});
