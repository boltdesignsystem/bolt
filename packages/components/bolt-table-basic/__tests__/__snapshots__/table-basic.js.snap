// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Table Basic Component Prop - borderless: true 1`] = `
<table class="c-bolt-table-basic c-bolt-table-basic--borderless c-bolt-table-basic--side-headers">
  <tbody>
    <tr>
      <th scope="row">
        Row 1
      </th>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <th scope="row">
        Row 2
      </th>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <th scope="row">
        Row 3
      </th>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Bolt Table Basic Component Prop - caption 1`] = `
<table class="c-bolt-table-basic">
  <caption>
    This is a caption
  </caption>
  <tbody>
    <tr>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Bolt Table Basic Component Prop - fixed_width_columns items: first 1`] = `
<table class="c-bolt-table-basic c-bolt-table-basic--fixed-width-columns--first">
  <thead>
    <tr>
      <th scope="col">
        Column 1
      </th>
      <th scope="col">
        Column 2
      </th>
      <th scope="col">
        Column 3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Bolt Table Basic Component Prop - fixed_width_columns items: first-two 1`] = `
<table class="c-bolt-table-basic c-bolt-table-basic--fixed-width-columns--first-two">
  <thead>
    <tr>
      <th scope="col">
        Column 1
      </th>
      <th scope="col">
        Column 2
      </th>
      <th scope="col">
        Column 3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Bolt Table Basic Component Prop - fixed_width_columns items: none 1`] = `
<table class="c-bolt-table-basic c-bolt-table-basic--fixed-width-columns--none">
  <thead>
    <tr>
      <th scope="col">
        Column 1
      </th>
      <th scope="col">
        Column 2
      </th>
      <th scope="col">
        Column 3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Bolt Table Basic Component Prop - fixed_width_columns items: second 1`] = `
<table class="c-bolt-table-basic c-bolt-table-basic--fixed-width-columns--second">
  <thead>
    <tr>
      <th scope="col">
        Column 1
      </th>
      <th scope="col">
        Column 2
      </th>
      <th scope="col">
        Column 3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Bolt Table Basic Component Prop - fixed_width_table: true 1`] = `
<table class="c-bolt-table-basic c-bolt-table-basic--fixed-width-table">
  <thead>
    <tr>
      <th scope="col">
        Column 1
      </th>
      <th scope="col">
        Column 2
      </th>
      <th scope="col">
        Column 3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Bolt Table Basic Component Prop - format items: numeric 1`] = `
<table class="c-bolt-table-basic c-bolt-table-basic--numeric c-bolt-table-basic--side-headers">
  <thead>
    <tr>
      <td>
      </td>
      <th scope="col">
        Column 1
      </th>
      <th scope="col">
        Column 2
      </th>
      <th scope="col">
        Column 3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th scope="row">
        Row 1
      </th>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <th scope="row">
        Row 2
      </th>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <th scope="row">
        Row 3
      </th>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <th>
        Footer
      </th>
      <td>
        FC1
      </td>
      <td>
        FC2
      </td>
      <td>
        FC3
      </td>
    </tr>
  </tfoot>
</table>
`;

exports[`Bolt Table Basic Component Prop - format items: regular 1`] = `
<table class="c-bolt-table-basic c-bolt-table-basic--side-headers">
  <thead>
    <tr>
      <td>
      </td>
      <th scope="col">
        Column 1
      </th>
      <th scope="col">
        Column 2
      </th>
      <th scope="col">
        Column 3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th scope="row">
        Row 1
      </th>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <th scope="row">
        Row 2
      </th>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <th scope="row">
        Row 3
      </th>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <th>
        Footer
      </th>
      <td>
        FC1
      </td>
      <td>
        FC2
      </td>
      <td>
        FC3
      </td>
    </tr>
  </tfoot>
</table>
`;

exports[`Bolt Table Basic default 1`] = `
<table class="c-bolt-table-basic">
  <tbody>
    <tr>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Bolt Table Basic footer table basic 1`] = `
<table class="c-bolt-table-basic c-bolt-table-basic--side-headers">
  <thead>
    <tr>
      <td>
      </td>
      <th scope="col">
        Column 1
      </th>
      <th scope="col">
        Column 2
      </th>
      <th scope="col">
        Column 3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th scope="row">
        Row 1
      </th>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <th scope="row">
        Row 2
      </th>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <th scope="row">
        Row 3
      </th>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <th>
        Footer
      </th>
      <td>
        FC1
      </td>
      <td>
        FC2
      </td>
      <td>
        FC3
      </td>
    </tr>
  </tfoot>
</table>
`;

exports[`Bolt Table Basic side header table basic 1`] = `
<table class="c-bolt-table-basic c-bolt-table-basic--side-headers">
  <tbody>
    <tr>
      <th scope="row">
        Row 1
      </th>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <th scope="row">
        Row 2
      </th>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <th scope="row">
        Row 3
      </th>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Bolt Table Basic top header table basic 1`] = `
<table class="c-bolt-table-basic">
  <thead>
    <tr>
      <th scope="col">
        Column 1
      </th>
      <th scope="col">
        Column 2
      </th>
      <th scope="col">
        Column 3
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        R1C1
      </td>
      <td>
        R1C2
      </td>
      <td>
        R1C3
      </td>
    </tr>
    <tr>
      <td>
        R2C1
      </td>
      <td>
        R2C2
      </td>
      <td>
        R2C3
      </td>
    </tr>
    <tr>
      <td>
        R3C1
      </td>
      <td>
        R3C2
      </td>
      <td>
        R3C3
      </td>
    </tr>
  </tbody>
</table>
`;
